trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/DataProviderService.java

```diff
@@ -1302,6 +1302,11 @@ public Response createProvider(
                 return errorResponse;
             }
 
+            IDataProviderDescriptor parentDescriptor = getDescriptor(experiment, outputId);
```

Comment:
line 1310 has the same effect. I don't think it's needed. You could update the error message below to add the output ID to it the error.

Reply:
In an earlier version of the TC PR, the factories weren't being filtered-out at all so this was necessary to prevent creation of derived provider from a hidden configurator.

In the latest TC PR, factories can be filtered-out and would prevent the creation, but it only works when the hiding configuration applies to all trace types.

So I believe that this is still necessary when the hiding configuration is only for a specific trace type.

Reply:
It's fine like that. It's inline for other custom data provider endpoints.